draft-gpew-priv-ppm.md

```diff
@@ -672,247 +677,377 @@ independently.
+   and is therefore a replay. An aggregator may also choose to mark an
+   input share as invalid with the  error `report-dropped` under the conditions
+   prescribed in {{anti-replay}}. The helper also checks to see if the report
+   has never been aggregated but is contained by a batch that has been collected.
 
-The helper state is an optional parameter of an aggregate request that the
-helper can use to carry state across requests. At least part of the state will
-usually need to be encrypted in order to protect user privacy. However, the
-details of precisely how the state is encrypted and the information that it
-carries is up to the helper implementation.
+If both checks succeed, the input share is not marked as invalid.
 
-### Aggregate Share Request {#aggregate-share-request}
+#### Input Share Preparation {#input-share-prep}
 
-Once the aggregators have verified at least as many reports as required for the
-PPM task, the leader issues an "aggregate-share request" to each helper. The
-helper responds to this request by extracting its aggregate share from its state
-and encrypting it under the collector's HPKE public key.
+Input share preparation consists of running the preparation-state initialization
+algorithm for the VDAF associated with the task and computes the first state transition.
+This produces three possible values:
 
-[OPEN ISSUE: consider updating the `checksum` algorithm to not permit
-collisions]
+1. An error, in which case the input report share is marked as invalid.
+1. An output share, in which case the aggregator stores the output share for future collection
+   as described in {{collect-flow}}.
+1. An initial VDAF state and preparation message, denoted `(prep_state, prep_msg)`.
 
-First, the leader computes a checksum over the set of output shares included in
-the batch window. The checksum is computed by taking the SHA256 hash of each
-nonce from the client reports included in the aggregation, then combining the
-hash values with a bitwise-XOR operation.
+Each aggregator runs this procedure for a given input share with corresponding nonce
+as follows:
 
-Then, for each aggregator endpoint `[aggregator]` in the parameters associated
-with `CollectReq.task_id` (see {{pa-collect}}) except its own, the leader sends
-a POST request to `[aggregator]/aggregate_share` with the following message:
+~~~
+prep_state = VDAF.prep_init(vdaf_verify_param, agg_param, nonce, input_share)
+out = VDAF.prep_next(prep_state, None)
+~~~
+
+`vdaf_verify_param` is the public VDAF parameter, and `agg_param` is the opaque aggregation
+parameter. If either step fails, the aggregator marks the report as invalid with error
+`vdaf-prep-error`.
+
+Otherwise, the value `out` is interpreted as follows. If the VDAF is 0-round, then `out`
+is the aggregator's output share. Otherwise, if the VDAF consists of one round or more,
+then the aggregator interprets `out` as the pair `(prep_state, prep_msg)`.
```

Comment:
```suggestion
Otherwise, the value `out` is interpreted as follows. If this is the last round of the VDAF,
then `out` is the aggregator's output share. Otherwise, `out` is the pair `(prep_state, prep_msg)`.
```
To match the corresponding paragraph describing the leader's behavior on L952.